### ISSUE TITLE
MNIST examples low accuracy

### ISSUE THREAD

**Issue by kkosteck** [BindsNET/bindsnet#553]
Running examples shows accuracy over 96% on training set, but after training, accuracy is on average around 10% with testing set.
I checked this for: [batch_eth_mnist.py](https://github.com/BindsNET/bindsnet/blob/master/examples/mnist/batch_eth_mnist.py), [eth_mnist.py](https://github.com/BindsNET/bindsnet/blob/master/examples/mnist/eth_mnist.py), [supervised_mnist.py](https://github.com/BindsNET/bindsnet/blob/master/examples/mnist/supervised_mnist.py)

Supervised mnist was checked by me with a lot of different values of parameters, but with same results. Every other example I run with default parameters.

**Comment by Hananel-Hazan** [comment]
Thank you for letting us know. 

The default parameter in the BindsNET script can provide decent performance, but it is not by all mean the top performance. To find the top performance one needs to run exploratory hyper-parameters.

Here are the performance on the following scripts:
eth_mnist.py
```
Train:
All activity accuracy: 75.95 (last), 63.96 (average), 79.44 (best)
Proportion weighting accuracy: 76.77 (last), 64.59 (average), 79.99 (best)
Test:
All activity accuracy: 0.77
Proportion weighting accuracy: 0.77
```

batch_eth_mnist.sh 
```
Train performance :
All activity accuracy: 77.33 (last), 66.90 (average), 79.66 (best)
Proportion weighting accuracy: 78.45 (last), 68.46 (average), 81.03 (best)
Test performance:
All activity accuracy: 0.79
Proportion weighting accuracy: 0.81 
```

SOM_LM-SNNs
```
Test:
All activity accuracy: 0.86
Proportion weighting accuracy: 0.86 
All activity accuracy: 100.00 (last), 99.62 (average), 100.00 (best)
Proportion weighting accuracy: 100.00 (last), 99.60 (average), 100.00 (best)
Train:
All activity accuracy: 84.00 (last), 74.07 (average), 88.80 (best)
Proportion weighting accuracy: 84.00 (last), 74.08 (average), 88.80 (best)

```
supervised_mnist
```
Train:
All activity accuracy: 75.95 (last), 63.96 (average), 79.44 (best)
Proportion weighting accuracy: 76.77 (last), 64.59 (average), 79.99 (best)
Test: 0.69
```

**Comment by kkosteck** [comment]
I understand that example parameters are not Hood enough to give top performance accuracy, but I don't understand why there is 99% accuracy on training set and only 10% on testing set. IT seems like with testing set it is just random choose, because MNIST has 10 classes and probability to pick one is exactly 10% like with testing set. Maybe there is some problem with my environment, but I was running this examples on google colab notebook with same results

**Comment by Hananel-Hazan** [comment]
Since I can't reproduce your reported accuracy numbers, I suggest you to be more specific on what exactly your are running, what parameters you are using and what is your environment of testing you are using.
